packages/liveblocks-core/src/client.ts

```diff
@@ -223,15 +223,17 @@ export function createClient(options: ClientOptions): Client {
     // TODO: Expose a way to clear these
     window.addEventListener("online", () => {
       for (const [, room] of rooms) {
-        room.__internal.onNavigatorOnline();
+        room.__internal.simulate.navigatorOnline();
       }
     });
   }
 
   if (typeof document !== "undefined") {
     document.addEventListener("visibilitychange", () => {
-      for (const [, room] of rooms) {
-        room.__internal.onVisibilityChange(document.visibilityState);
+      if (document.visibilityState === "visible") {
```

Comment:
Another subtle change: I've renamed the `onVisibilityChange` to `windowGotFocus` here, meaning it should only get called when the window is refocused, and no longer when the window loses focus. This check used to live in the handler, but now it lives as a guard here, where the event will be fired. This will match more closely with the events from our [target state diagram](https://stately.ai/registry/editor/e6c62b06-19f2-44fe-9f9a-d69472d262fd?machineId=cee0b455-b631-4be9-a614-07789f711b01&mode=Simulate).